modules/acquia_cms_headless/modules/acquia_cms_headless_ui/acquia_cms_headless_ui.module

```diff
@@ -29,6 +31,20 @@ function acquia_cms_headless_ui_help($route_name) {
   }
 }
 
+/**
+ * Implements hook_node_update().
+ */
+function acquia_cms_headless_ui_node_update(NodeInterface $node) {
+  $url = Url::fromRoute('entity.node.headless_preview', ['node' => $node->id()]);
+  // Check for destination query param.
+  $current_request = \Drupal::service('request_stack')->getCurrentRequest();
+  // If not set, then redirect to site-preview.
+  if (!$current_request->query->get('destination')) {
+    $response = new RedirectResponse($url->toString());
+    $response->send();
```

Comment:
How is this going to respond in (a) bulk update scenarios and (b) over CLI?

Reply:
This feels like an action we should take on the form rather than on hook_node_update.
This module also has a number or route overrides that maybe worth reviewing. Perhaps the better approach is to set the /node/<nid>/latest route to use entity.node.headless_preview controller?